tests/integration/pilot/bindpodip/bindpodip_test.go

```diff
@@ -0,0 +1,231 @@
+
+			var client echo.Instance
+			var instance echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&instance, echo.Config{
+					Service:   "server",
+					Namespace: ns,
+					Subsets:   []echo.SubsetConfig{{Annotations: echo.NewAnnotations().Set(echo.BindPodIPPorts, "6666,7777")}},
+					Pilot:     p,
+					Galley:    g,
+					Ports: []echo.Port{
+						{
+							Name:         "http5",
+							Protocol:     protocol.HTTP,
+							ServicePort:  5555,
+							InstancePort: 5555,
+						},
+						{
+							Name:         "http6",
+							Protocol:     protocol.HTTP,
+							ServicePort:  6666,
+							InstancePort: 6666,
+						},
+						{
+							Name:         "http7",
+							Protocol:     protocol.HTTP,
+							ServicePort:  7777,
+							InstancePort: 7777,
+						},
+					},
+				}).With(&client, echo.Config{
+				Service:   "client",
+				Namespace: ns,
+				Galley:    g,
+				Pilot:     p,
+				Ports: []echo.Port{
+					{
+						Name:     "grpc",
+						Protocol: protocol.GRPC,
+					},
+				},
+			}).
+				BuildOrFail(t)
+
+			clientWorkload := getWorkload(client, t)
+			serverWorkload := getWorkload(instance, t)
+			sidecar := serverWorkload.Sidecar()
+			podIP := strings.Split(sidecar.NodeID(), "~")[1]
+			expectedPortToEndpointMap := map[string]string{
+				"5555": "127.0.0.1",
+				"6666": podIP,
+				"7777": podIP,
+			}
+			configDump, err := sidecar.Config()
+			if err != nil {
+				t.Fatal(err)
+			}
+			wrapper := configdump.Wrapper{ConfigDump: configDump}
+			dcd, err := wrapper.GetClusterConfigDump()
+			if err != nil {
+				t.Fatal(err)
+			}
+			for _, cluster := range dcd.DynamicActiveClusters {
+				clusterTyped := &xdsapi.Cluster{}
+				err = ptypes.UnmarshalAny(cluster.Cluster, clusterTyped)
+				if err != nil {
+					t.Fatal(err)
+				}
+				for port, bind := range expectedPortToEndpointMap {
+					if strings.HasPrefix(clusterTyped.Name, fmt.Sprintf("%s|%s", model.TrafficDirectionInbound, port)) {
+						ep, ok := clusterTyped.LoadAssignment.Endpoints[0].LbEndpoints[0].HostIdentifier.(*endpoint.LbEndpoint_Endpoint)
+						if !ok {
+							t.Errorf("failed convert endpoint")
+							break
+						}
+						sa, ok := ep.Endpoint.Address.Address.(*core.Address_SocketAddress)
+						if !ok {
+							t.Errorf("failed convert address")
+						}
+						addr := sa.SocketAddress.Address
+						if addr != bind {
+							t.Errorf("port %s is expected to be bind to %s but got %s", port, bind, addr)
+						}
+						break
+					}
+				}
+			}
+
+			for port := range expectedPortToEndpointMap {
+				name := fmt.Sprintf("client->%s:%s", instance.Config().Service, port)
+				host := fmt.Sprintf("%s:%s", serverWorkload.Address(), port)
+				request := &epb.ForwardEchoRequest{
+					Url:   fmt.Sprintf("http://%s/", host),
+					Count: 1,
+					Headers: []*epb.Header{
+						{
+							Key:   "Host",
+							Value: host,
+						},
+					},
+				}
+				t.Run(name, func(t *testing.T) {
+					retry.UntilSuccessOrFail(t, func() error {
+
```

Comment:
This test doesn't really do anything, does it? Because echo is listeneing on `0.0.0.0` already. So even if you did not make ANY changes here except add the test, it would have passed?

Reply:
you are right. 
Will edit echo server to listen on podIP, and delete the config dump test case.